--- a/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dns.js
+++ b/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dns.js
@@ -442,6 +442,9 @@ return view.extend({
 		so.rmempty = false;
 		so.datatype = 'ipaddr("nomask")';
 
+		so = ss.option(form.Value, 'comments', _('Comments'));
+		so.rmempty  = true;
+
 		var ipaddrs = {};
 
 		Object.keys(hosts).forEach(function(mac) {

--- a/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ b/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -308,6 +308,10 @@ return view.extend({
 			_('Authoritative'),
 			_('This is the only DHCP server in the local network.'));
 
+		s.taboption('general', form.Flag, 'dns_redirect',
+			_('DNS Redirect'),
+			_('Force redirect all local DNS queries to DNSMasq, a.k.a. DNS Hijacking.'));
+
 		s.taboption('general', form.Value, 'domain',
 			_('Local domain'),
 			_('Local domain suffix appended to DHCP names and hosts file entries.'));
